Sources/Plasma/PubUtilLib/plAudioCore/plSoundBuffer.cpp

```diff
@@ -82,28 +82,30 @@ static plAudioFileReader *CreateReader( bool fullpath, const plFileName &filenam
 
 void plSoundPreloader::Run()
 {
-    hsTArray<plSoundBuffer*> templist;
+    std::vector<plSoundBuffer*> templist;
 
     while (fRunning)
     {
         {
             hsLockGuard(fCritSect);
-            while (fBuffers.GetCount())
+            while (!fBuffers.empty())
             {
-                templist.Append(fBuffers.Pop());
+                templist.emplace_back(fBuffers.back());
+                fBuffers.pop_back();
```

Comment:
Doesn't the STL have a more optimized way to copy the contents of one container to another (in reverse order)?